poseidon/src/sponge.rs

```diff
@@ -156,13 +153,56 @@ where
     }
 
     pub fn squeeze(&mut self, num_limbs: usize) -> P::ScalarField {
-        P::ScalarField::from_repr(pack(&self.squeeze_limbs(num_limbs)))
+        P::ScalarField::from_bigint(pack(&self.squeeze_limbs(num_limbs)))
             .expect("internal representation was not a valid field element")
     }
 }
 
-impl<P: SWModelParameters, SC: SpongeConstants>
-    FqSponge<P::BaseField, GroupAffine<P>, P::ScalarField> for DefaultFqSponge<P, SC>
+// Debugging macros -- these only insert code when non-release build and
+// "debug_sponge" feature is enabled.
+macro_rules! debug_sponge {
```

Comment:
I think this can be removed. It comes from the cherry-picking. Reviewers should check.